app/views/my/page_layout.html.erb

```diff
@@ -88,30 +88,57 @@ See doc/COPYRIGHT.rdoc for more details.
   <li><%= link_to l(:button_back), {:action => 'page'}, :class => 'icon icon-cancel' %></li>
 <% end %>
 
+<% content_for :toolbar do %>
+    <li class="toolbar-item">
+      <%= styled_form_tag({:action => "add_block"}, :id => "block-form") do %>
+        <%= styled_select_tag 'block', "<option>--#{t(:button_add)}--</option>".html_safe + options_for_select(@block_options),
+              :id => "block-select",
+              class: '-small' %>
+      <% end %>
+    </li>
+    <li class="toolbar-item">
+      <%= link_to_remote l(:button_add),
```

Comment:
This would require some extensive changes to the way the my page implementation works. this PR only deals with some styling issues regarding the module.

I would delay changing the implementation until we move the plugin to core.